### PR TITLE
upgrade chia-rs to streamable support

### PR DESCRIPTION
**Pull request by arvidn**
Update `chia_rs` to the latest release which includes Streamable serialization and deserialization support for `PySpend` and `PySpendBundleConditions` classes, which are the output from running a generator program and parsing and validating the conditions. (those classes have changed names to remove the `Py` prefix and gained type-hints).

In addition to supporting serialization and deserialization, it also adds support to `to_json_dict()`, which we use to print these types (mostly in tests, but also in error logging).

The reason to serialize and deserialize these in rust instead of python is that they are expensive enough to show up in profiles of the main process. The main thread is especially sensitive to using CPU since it's doing all serial work and holds up the parallel block validation in the worker processes.

This change only makes a small impact on wall-clock time of the benchmarks. This is because the main process is dominated by disk I/O. Improving the disk performance is a bigger long-term project that will have to be addressed separately. This patch is primarily concerned with reducing the CPU usage.

Wall clock time:
| test | before | after | after/before |
| --- | --- | --- | --- |
| M1, SSD, sync | 114.04s | 109.73s | 0.96 |

CPU time (`receive_block_batch()`):
| test | before | after | after/before |
| --- | --- | --- | --- |
| M1, SSD, sync | 21.37% | 17.11% | 0.80 |

# Profiles

(`pre_validate_blocks_multiprocessing()` is where we deserialize the generator results from the worker processes)

before:
slow-batch-00448

after:

slow-batch-00448

### CODE REVIEW

File: chia/util/streamable.py [hunk 1]
```diff
@@ -66,6 +67,7 @@ class DefinitionError(StreamableError):
     "G2Element",
     "Program",
     "SerializedProgram",
+    "PySpendBundleConditions",
```
Comment (arvidn) [inline chia/util/streamable.py:70]: I believe this is wrong. It was an earlier attempt I made

File: chia/types/spend_bundle_conditions.py [hunk 2]
```diff
@@ -1,30 +1,3 @@
-from dataclasses import dataclass
-from typing import List, Optional, Tuple
+from chia_rs import Spend, SpendBundleConditions
```
Comment (arvidn) [inline chia/types/spend_bundle_conditions.py:1]: the type stubs for these classes are here: https://github.com/Chia-Network/chia_rs/blob/main/wheel/chia_rs.pyi

**Comment by arvidn** [comment]
I ran through a successful sync-from-scratch:

```
$ python ./tools/test_full_sync.py run --profile ~/chia-backup/db/blockchain_v2_mainnet.sqlite
PID: 2009491
Chia directory /tmp/tmp854fo5t9
Can't find private CA, creating a new one in /tmp/tmp854fo5t9 to generate TLS certificates

To see your keys, run 'chia keys show --show-mnemonic-seed'

height 100000 0.01 s/block
height 200000 0.01 s/block
height 300000 0.02 s/block
height 400000 0.02 s/block
height 500000 0.04 s/block
height 600000 0.02 s/block
height 700000 0.03 s/block
height 800000 0.03 s/block
height 900000 0.04 s/block
height 1000000 0.05 s/block
height 1100000 0.02 s/block
height 1200000 0.03 s/block
height 1300000 0.04 s/block
height 1400000 0.02 s/block
height 1500000 0.02 s/block
height 1600000 0.12 s/block
height 1700000 0.09 s/block
height 1772864 0.02 s/block   closing full node
```

**Review by richardkiss — APPROVED** [review]
This seems reasonable.

Note that `maturin` now supports mixed-mode wheels that have python alongside the rust-based shared object that's loaded. You might consider investigating that, as it might allow you to push a lot of these wrapping diffs into python in the `chia_rs` wheel (you'd have to push the `bytes32` definition into `chia_rs`, but that's might not be completely terrible).

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**